pkg/agent/openflow/network_policy.go

```diff
@@ -1313,3 +1336,73 @@ func (c *client) ReassignFlowPriorities(updates map[uint16]uint16, table binding
 	}
 	return nil
 }
+
+func parseDropFlow(flow string) (uint32, types.RuleMetric) {
+	// example format:
+	// table=101, n_packets=9, n_bytes=666, priority=200,ip,reg0=0x100000/0x100000,reg3=0x5 actions=drop
+	segs := strings.Split(flow, ",")
+	m := types.RuleMetric{}
+	pkts, _ := strconv.ParseUint(segs[1][strings.Index(segs[1], "=")+1:], 10, 32)
+	m.Packets = pkts
+	bytes, _ := strconv.ParseUint(segs[2][strings.Index(segs[2], "=")+1:], 10, 32)
+	m.Bytes = bytes
+	id, _ := strconv.ParseUint(segs[6][strings.Index(segs[6], "0x")+2:strings.Index(segs[6], " ")], 16, 32)
+	return uint32(id), m
+}
+
+func parseAllowFlow(flow string) (uint32, types.RuleMetric) {
+	// example format:
+	// table=101, n_packets=0, n_bytes=0, priority=200,ct_state=-new,ct_label=0x1/0xffffffff,ip actions=goto_table:105
+	segs := strings.Split(flow, ",")
+	m := types.RuleMetric{}
+	pkts, _ := strconv.ParseUint(segs[1][strings.Index(segs[1], "=")+1:], 10, 32)
+	m.Packets = pkts
+	if strings.Contains(segs[4], "+") {
+		m.Sessions = pkts
+	}
+	bytes, _ := strconv.ParseUint(segs[2][strings.Index(segs[2], "=")+1:], 10, 32)
+	m.Bytes = bytes
+	idRaw := segs[5][strings.Index(segs[5], "0x")+2 : strings.Index(segs[5], "/")]
+	if len(idRaw) > 8 { // only 32 bits are valid.
+		idRaw = idRaw[:len(idRaw)-8]
+	}
+	id, _ := strconv.ParseUint(idRaw, 16, 32)
+	return uint32(id), m
+}
+
+func parseMetricFlow(flow string) (uint32, types.RuleMetric) {
+	if strings.Contains(flow, "reg0") {
+		return parseDropFlow(flow)
+	} else {
+		return parseAllowFlow(flow)
+	}
+}
+
+func (c *client) NetworkPolicyMetrics() map[uint32]*types.RuleMetric {
+	result := map[uint32]*types.RuleMetric{}
+	egressFlows, _ := ovsctl.NewClient(c.nodeConfig.OVSBridge).DumpTableFlows(uint8(EgressMetricTable))
+	for _, flow := range egressFlows {
+		if strings.Contains(flow, "priority=0") {
+			continue
+		}
+		ruleID, metric := parseMetricFlow(flow)
+		if accMetric, ok := result[ruleID]; ok {
+			accMetric.Merge(metric)
+		} else {
+			result[ruleID] = &metric
+		}
+	}
+	ingressFlows, _ := ovsctl.NewClient(c.nodeConfig.OVSBridge).DumpTableFlows(uint8(IngressMetricTable))
```

Comment:
@weiqiangt actually I wasn't suggesting making it a field in the client. I was just suggesting having: `ovsctlClient := ovsctl.NewClient(c.nodeConfig.OVSBridge)` at the start of the function instead of building 2 clients.

Reply:
I also agree that there is no good reason to make it a field since we only use it here.